aten/src/ATen/native/Histogram.cpp

```diff
@@ -6,117 +6,181 @@
-                " input tensor and weight tensor should have the same shape, but got input(", input.sizes(), ")",
-                ", and weight(", weight.value().sizes(), ")");
+        auto input_sizes = input.sizes().vec();
+        input_sizes.pop_back();
+
+        auto weight_sizes = weight.value().sizes().vec();
+        if (weight_sizes.empty())
+            weight_sizes = {1}; // correctly handle scalars
+
+        TORCH_CHECK(input_sizes == weight_sizes, "torch.histogram: if weight tensor is provided,"
+                " it should have the same shape as the input tensor excluding its last dimension, but got input ",
+                input.sizes(), " and weight ", weight.value().sizes());
     }
 }
 
 /* Checks properties of output tensors hist and bin_edges, then resizes them.
  */
-void histogram_prepare_out(const Tensor& input, int64_t bin_ct,
-        const Tensor& hist, const Tensor& bin_edges) {
+void histogramdd_prepare_out(const Tensor& input, const std::vector<int64_t>& bin_ct,
+        const Tensor& hist, const TensorList& bin_edges) {
+    const int64_t D = input.size(-1);
+
+    TORCH_INTERNAL_ASSERT(int64_t(bin_ct.size()) == D);
+    TORCH_INTERNAL_ASSERT(int64_t(bin_edges.size()) == D);
+
     TORCH_CHECK(input.dtype() == hist.dtype(), "torch.histogram: input tensor and hist tensor should",
             " have the same dtype, but got input ", input.dtype(), " and hist ", hist.dtype());
 
-    TORCH_CHECK(input.dtype() == bin_edges.dtype(), "torch.histogram: input tensor and bin_edges tensor should",
-            " have the same dtype, but got input ", input.dtype(), " and bin_edges ", bin_edges.dtype());
+    for (int64_t dim = 0; dim < D; dim++) {
+        TORCH_CHECK(input.dtype() == bin_edges[dim].dtype(), "torch.histogram: input tensor and bin_edges tensor should",
+                " have the same dtype, but got input ", input.dtype(), " and bin_edges ", bin_edges[dim].dtype(),
+                " for dimension ", dim);
 
-    TORCH_CHECK(bin_ct > 0,
-            "torch.histogram(): bins must be > 0, but got ", bin_ct);
+        TORCH_CHECK(bin_ct[dim] > 0,
+                "torch.histogram(): bins must be > 0, but got ", bin_ct[dim], " for dimension ", dim);
+
+        at::native::resize_output(bin_edges[dim], bin_ct[dim] + 1);
+    }
 
     at::native::resize_output(hist, bin_ct);
+}
+void histogramdd_prepare_out(const Tensor& input, TensorList bins,
+        const Tensor& hist, const TensorList& bin_edges) {
+    std::vector<int64_t> bin_ct(bins.size());
+    std::transform(bins.begin(), bins.end(), bin_ct.begin(), [](Tensor t) { return t.numel() - 1; });
+    histogramdd_prepare_out(input, bin_ct, hist, bin_edges);
+}
+
+template<typename scalar_t>
+void infer_bin_edges_from_input(const Tensor& input, const int64_t D,
+        std::vector<double> &leftmost_edges, std::vector<double> &rightmost_edges) {
+    Tensor min, max;
+    std::tie(min, max) = aminmax(input, 0);
```

Comment:
Would you add a comment to how this call to aminmax works? dim=0, which I think is what's going on here, seems odd if just reading the code